File: src/Domain/Emails/Models/CustomEmail.cs
using System.ComponentModel.DataAnnotations;
using Utils.Helpers;
using Utils.Interfaces;

namespace Domain.Emails.Models
{
    public class CustomEmail : EmailBase
    {
        public CustomEmail(
            EmailContent emailContent,
            IViewRenderer renderer,
            IBaseUrls url)
            : base(renderer, url)
        {
            emailContent.Subject.ThrowIfNull("subject", nameof(emailContent.Subject));
            Content = emailContent.Body;
            Subject = emailContent.Subject;
            From = emailContent.From;

            foreach (var recipient in emailContent.Recipients)
            {
                Recipients.Add(recipient);
            }

            foreach (var copy in emailContent.Cc)
            {
                Cc.Add(copy);
            }

            foreach (var hiddenCopy in emailContent.HiddenCc)
            {
                HiddenCc.Add(hiddenCopy);
            }
        }

        protected override string ViewPath => "/Content/EmailTemplates/CustomEmail.cshtml";

        [Required]
        protected override string From { get; }

        [Required]
        protected override string Subject { get; }

        public string Content { get; }

        public string EmailTitle => Subject;
    }
}

File: src/Frontend/src/app/modules/admin/components/emails/custom-email/additional-classes/custom-email.ts
import { IEmailPreview } from '@models/interfaces';
import Assertion from '@shared/validation/assertion';

export class CustomEmail implements IEmailPreview {
  readonly from: string;
  readonly recipients: Array<string>;
  readonly cc: Array<string>;
  readonly hiddenCc: Array<string>;
  readonly subject: string;
  readonly body: string;

  constructor(
    from: string,
    recipients: Array<string>,
    cc: Array<string>,
    hiddenCc: Array<string>,
    subject: string,
    body: string
  ) {
    Assertion.notNull(from, 'from');
    Assertion.notNull(recipients, 'recipients');
    Assertion.notNull(subject, 'subject');

    this.from = from;
    this.recipients = recipients;
    this.cc = cc;
    this.hiddenCc = hiddenCc;
    this.subject = subject;
    this.body = body;
  }
}
